Clean-up anchors and definitions

alexshalamov opened the pull request:
Based on feedback from Marcos, this PR introduces following changes:

- Removes unnecessary `<code></code>` tags around definitions
- Adds data-dfn-for and data-link-for for sections to avoid duplication
- Replaces 'for' attribute with 'data-dfn-for'

Commit:
Clean-up anchors and definitions

Based on feedback from Marcos, this PR introduces following changes:

- Removes unnecessary <code></code> tags around definitions
- Adds data-dfn-for and data-link-for for sections to avoid duplications
- Replaces 'for' attribute with 'data-dfn-for'

alexshalamov commented:
@marcoscaceres this PR addresses comments that you gave in #108

marcoscaceres commented on index.html:
```diff
@@ -131,12 +131,12 @@
         <li>
-          <code><dfn><a href=
-          'https://html.spec.whatwg.org/multipage/browsers.html#window'>Window</a></dfn></code>
+          <dfn><a href=
```
We need to update these (I mislead you a bit, here is ok to use code, then ReSpec automatically "codes" any `a` that xref this)... we can do these separately tho unless you feel like doing them now. 

```HTML
<code><dfn data-cite="!HTML/multipage/browsers.html#window">Window</dfn></code>
```

See: https://github.com/w3c/respec/wiki/data--cite

marcoscaceres commented on index.html:
```diff
@@ -231,45 +231,44 @@
-          When the <code><dfn for='ScreenOrientation'>lock</dfn>()</code>
-          method is invoked, the <a>user agent</a> MUST run the <a>apply an
-          orientation lock</a> steps to the <a>responsible document</a> using
+          When the <dfn>lock</dfn>() method is invoked,
```
Nit: we now support `<dfn>lock()</dfn>` (i.e., parens inside)

marcoscaceres commented on index.html:
```diff
@@ -212,13 +212,13 @@
         </pre>
         <p>
-          The <code><dfn for='Screen'>orientation</dfn></code> object is an
+          The <dfn for='Screen'>orientation</dfn> object is an
```
Nit: for attribute is not supported.

marcoscaceres commented on index.html:
```diff
@@ -231,45 +231,44 @@
-          When the <code><dfn for='ScreenOrientation'>unlock</dfn>()</code>
-          method is invoked, the <a>user agent</a> MUST run the steps to
-          <a>lock the orientation</a> of the <a>responsible document</a> to the
+          When the <dfn>unlock</dfn>() method is
```
as above.

marcoscaceres commented on index.html:
```diff
@@ -231,45 +231,44 @@
-          handler event type</a> is <code>change</code>.
+          The <dfn>onchange</dfn> attribute is an
+          <a>event handler</a> whose corresponding <a>event handler event type</a>
+          is <code>change</code>.
```
we should probably write `<code>"change"</code>`... if we defined "change", we should probably link to it... can't remember tho.

marcoscaceres commented on index.html:
```diff
@@ -231,45 +231,44 @@
+          is <code>change</code>.
         </p>
       </section>
       <section>
```
Please add `data-link-for="OrientationType" data-dfn-for="OrientationType"` attributes to this section.

marcoscaceres commented on index.html:
```diff
@@ -355,15 +354,13 @@
-            <code><dfn for='OrientationType'>landscape-primary</dfn></code>
-            or <code><dfn for=
-            'OrientationType'>landscape-secondary</dfn></code>.
+          <dfn data-dfn-for='OrientationType'>landscape-primary</dfn> or
```
Let's move the `data-dfn-for='OrientationType'` to the `section` element, then it will just work with `<dfn>`

marcoscaceres commented on index.html:
```diff
@@ -463,20 +460,20 @@
-                'OrientationLockType'>landscape-primary</dfn></code> or
-                <code><dfn for=
-                'OrientationLockType'>landscape-secondary</dfn></code>
+                <dfn data-dfn-for=
```
as above... then you don't need `data-dfn-for` here... ReSpec always looks for the nearest section with "data-dfn-for=".

marcoscaceres commented on index.html:
```diff
@@ -702,9 +698,9 @@
-              MUST be annotated with <code>process user orientation change
-              </code> when running the next step.
+              <a>lock</a>, the <a>task</a> MUST be annotated with
+              <code>process user orientation change</code> when running the
```
Nit: this `code` should probably have been an `<a>`

alexshalamov replied:
Fixed all comments except this one, as it needs definition or algo for `process user orientation change` term.

marcoscaceres reviewed (changes requested):
Nearly there!

Commit:
Update dependencies section and use data-dfn-for attributes

- Update deps section to use data-cite attribute
- Update broken links, e.g., "now visible algorithm"
- Use data-dfn-for for OrientationType and OrientationLockType definitions

marcoscaceres reviewed (approved):
Thanks again @alexshalamov for your work on this!
